src/lumigo_tracer/extension/extension.py

```diff
@@ -48,7 +52,8 @@ def shutdown(self):
             self._finish_previous_invocation(current_bandwidth)
 
     def _finish_previous_invocation(self, current_bandwidth: Optional[int]):
-        self.sampler.stop_sampling()
```

Comment:
there should be a single sampler.
Maybe you can change this sampler to collect two metrics!